compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -5550,19 +5573,19 @@ private STNode parseObjectMember(SyntaxKind nextTokenKind, STNode metadata) {
                 member = parseObjectMethodOrField(metadata, visibilityQualifier);
                 break;
             case REMOTE_KEYWORD:
-                member = parseObjectMethodOrField(metadata, STNodeFactory.createEmptyNode());
-                break;
             case FUNCTION_KEYWORD:
             case TRANSACTIONAL_KEYWORD:
-                member = parseObjectMethod(metadata, STNodeFactory.createEmptyNode(), STNodeFactory.createEmptyNode());
+            case RESOURCE_KEYWORD: // resource qualifier is not allowed but let it pass here and validate in
+                // parseFunctionQualifiers method
+                member = parseObjectMethod(metadata, new ArrayList<>());
                 break;
             default:
                 if (isTypeStartingToken(nextTokenKind)) {
                     member = parseObjectField(metadata, STNodeFactory.createEmptyNode());
                     break;
                 }
 
-                Solution solution = recover(peek(), ParserRuleContext.OBJECT_MEMBER_WITHOUT_METADATA);
+                Solution solution = recover(peek(), ParserRuleContext.OBJECT_MEMBER_WITHOUT_METADATA, metadata);
```

Comment:
Shall we also modify ```return parseObjectMember(solution.tokenKind);``` beneath. Otherwise we'll be re-parsing with the wrong method I think.

Reply:
U mean to rename the functions?

Reply:
renamed the functions

Reply:
not exactly the renaming of method. Earlier we had called the wrong overloaded method (with missing metadata arg) :)

Reply:
Yes, thanx for mentioning I have changed that.